### PR TITLE
Downgrade user-secrets CLI tool to use 1.0.0 runtime

### PR DESCRIPTION
**Pull request by natemcmaster**
This required splitting runtime testing into its own project.

Will split a rel/tools-rel-1.0.0-preview3 branch after this merges. cref #100 

cc @pranavkm @muratg

### CODE REVIEW

**Comment by natemcmaster** [comment]
Ping. Can someone review this?

File: src/Microsoft.Extensions.SecretManager.Tools/Program.cs [hunk 1]
```diff
@@ -224,18 +224,25 @@ private void PrintAll(IDictionary<string, string> secrets)
...
 
-        private void ProcessSecretFile(string projectPath, Action<IDictionary<string,string>> observer, bool persist = true)
+        private void ProcessSecretFile(string projectPath, Action<IDictionary<string, string>> observer, bool persist = true)
         {
             Logger.LogDebug(Resources.Message_Project_File_Path, projectPath);
             var secretsFilePath = PathHelper.GetSecretsPath(projectPath);
             Logger.LogDebug(Resources.Message_Secret_File_Path, secretsFilePath);
+
+            {
+                // workaround https://github.com/aspnet/Configuration/issues/478
```
Comment (pranavkm) [inline src/Microsoft.Extensions.SecretManager.Tools/Program.cs]: Could you file a bug for this and self assign? Easy to miss TODOs in code.

File: src/Microsoft.Extensions.SecretManager.Tools/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -3,9 +3,7 @@
 
 using System.Reflection;
 using System.Resources;
-using System.Runtime.CompilerServices;
 
-[assembly: InternalsVisibleTo("Microsoft.Extensions.SecretManager.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```
Comment (pranavkm) [inline src/Microsoft.Extensions.SecretManager.Tools/Properties/AssemblyInfo.cs]: Why?

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.Extensions.SecretManager.Tools/Program.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.SecretManager.Tools/Program.cs]: Good idea. Added https://github.com/aspnet/UserSecrets/issues/102

**Pull request merged**